services/rfq/relayer/service/handlers.go

```diff
@@ -378,8 +378,28 @@ func (r *Relayer) handleRelayLog(parentCtx context.Context, req *fastbridge.Fast
 // Step 6: ProvePosting
 //
 // This is the sixth step in the bridge process. Here we submit the claim transaction to the origin chain.
-func (q *QuoteRequestHandler) handleRelayCompleted(ctx context.Context, _ trace.Span, request reldb.QuoteRequest) (err error) {
-	// relays been completed, it's time to go back to the origin chain and try to prove
+func (q *QuoteRequestHandler) handleRelayCompleted(ctx context.Context, span trace.Span, request reldb.QuoteRequest) (err error) {
+	// first, ensure that enough confirmations have passed before proving
+	receipt, err := q.Dest.Client.TransactionReceipt(ctx, request.DestTxHash)
+	if err != nil {
+		return fmt.Errorf("could not get transaction receipt: %w", err)
+	}
+	currentBlockNumber := q.Origin.LatestBlock()
+	proveConfirmations, err := q.cfg.GetFinalityConfirmations(int(q.Dest.ChainID))
+	if err != nil {
+		return fmt.Errorf("could not get prove confirmations: %w", err)
+	}
+	span.SetAttributes(
+		attribute.Int("current_block_number", int(currentBlockNumber)),
+		attribute.Int("dest_block_number", int(receipt.BlockNumber.Int64())),
+		attribute.Int("prove_confirmations", int(proveConfirmations)),
+	)
+	if currentBlockNumber < request.BlockNumber+proveConfirmations {
+		span.AddEvent("not enough confirmations")
+		return nil
+	}
+
+	// relay has been finalized, it's time to go back to the origin chain and try to prove
```

Comment:
@dwasse we need to do a check similar to Guard's `isProveValid` here to handle the case where our relay tx was reorged.

In the event where we submitted a tx, witnessed a `BridgeRelayed` event, and later on this tx was reorged and ended up being reverted, we need a way for the Relayer to mark the tx as RelayRaceLost here (I'm not sure if we can safely assume that we will index the correct event with another relayer before that).

The current code will still get a valid receipt for `DestTxHash`, it's just the receipt will not have a corresponding `BridgeRelayed` event there.